pkg/registry/core/service/portallocator/operation.go

```diff
@@ -104,6 +128,10 @@ func (op *PortAllocationOperation) Allocate(port int) error {
 
 // Allocates a port, and record it for future rollback
 func (op *PortAllocationOperation) AllocateNext() (int, error) {
+	if op.dryRun {
```

Comment:
Suggestions
* Add this to the op.allocated array as well
* if op.allocated is non-empty, take the last entry and add 1
* 0 is a special port number. Maybe start at something around 2^15.

Reply:
Actually, is it even possible to cause this function to be called more than once when creating a service? If so we should test it, if not the second suggestion is probably not necessary.

Reply:
It is possible, I'll add a test case for this